language/ir-testsuite/tests/move/generics/phantom_params/struct_instantiation.mvir

```diff
@@ -0,0 +1,85 @@
+// This file contains tests checking that type arguments used in phantom position
```

Comment:
might be good to have negative versions of all of these tests.... thought I can't really see what those would be?

Maybe have some tests like `HasAbility<phantom T1, T2> has Ability` where `T2` does and does not have that ability. To show `T1` doesn't impact it. Thoughts?

Reply:
The negative test feels weird to me and I don't think it adds much here as not having the ability doesn't actually mean `T1` doesn't have an impact. Also, "negative bugs" are not too bad because fixing them is always going to be backward-compatible as you would be lifting restrictions. This may be the reason they feel weird to me.

Mixing phantom and non-phantom parameters make a lot of sense though. I'll start by adding that to the existing tests.

Reply:
well, I end up doing anyway...

Reply:
Here I was referring to negative tests as tests that throw errors. But it is also very important to test any sort of "negative bugs" resulting from a lack of precision in the analysis. The tests are more there to catch some sort of inconsistency that arises from a change, or to track features as they develop. In short, if they start passing, be afraid :P 

I'll take a look at the test changes